### PR TITLE
chore: improve code coverage

### LINKED ISSUE #348: Increase coverage to 80%
**Is your feature request related to a problem? Please describe.**
We need to increase test coverage of meteor to some threshold, 80% would be sufficient for now.

**Describe the solution you'd like**
Increase test coverage to 80%

### PR DESCRIPTION
**Pull request by GrayFlash**
closes #348

### CODE REVIEW

File: plugins/external/discover_test.go [hunk 1]
```diff
@@ -6,11 +6,15 @@ package plugins
 import (
 	"testing"
 
+	"github.com/odpf/meteor/registry"
 	"github.com/stretchr/testify/assert"
 )
 
 func TestDiscoverPlugins(t *testing.T) {
```
Comment (GrayFlash) [inline plugins/external/discover_test.go:13]: @mabdh I need some context arounf `plugins/external` directory, around its usecase, and requirements. Anything in docs/previous-PR/issue/discussion??

File: plugins/extractors/bigquery/profile_test.go [hunk 2]
```diff
@@ -1,3 +1,6 @@
+//go:build plugins
+// +build plugins
+
```
Comment (GrayFlash) [inline plugins/extractors/bigquery/profile_test.go:3]: Added plugins tags for tableau and bigquery tests, because we test extractors separately.

**Comment by GrayFlash** [comment]
@mabdh @StewartJingga  @scortier any suggestions for further improving the code coverage?? Most of the files with low coverage that I can see at the moment are the ones where we can't test a given set of functions with mocks or may require setting up a service (like bigquery, tableau, etc).

File: plugins/external/discover_test.go (hunk 1 above)
Comment (mabdh) [reply plugins/external/discover_test.go:13]: @GrayFlash I don't really aware of the docs about it, maybe @ravisuhag or @StewartJingga has more context with that.

**Comment by mabdh** [comment]
@GrayFlash this is already good enough to have 75 % code coverage.

Have you seen the coverage report here? https://coveralls.io/github/odpf/meteor?branch=improve-code-coverage
apart from plugins files there are several files that could be improved.

**Comment by GrayFlash** [comment]
@mabdh have worked on increasing the coverage across the `agent`, `registry` and `recipe` package, by adding tests for Validation functions as well. Please check it once.

**Review by mabdh — APPROVED** [review]

**Pull request merged**